src/main/java/com/github/packageurl/PackageURL.java

```diff
@@ -32,11 +32,13 @@
 import java.util.Collections;
 import java.util.Map;
 import java.util.Objects;
+import java.util.StringJoiner;
 import java.util.TreeMap;
 import java.util.function.IntPredicate;
 import java.util.stream.Collectors;
 import java.util.stream.IntStream;
-import org.jspecify.annotations.Nullable;
+
+import static java.util.Objects.requireNonNull;
```

Comment:
I am not sure about the reordering of imports. This happened automatically. The formatting plugin should fix anyway, I hope.